### PR TITLE
fix: add metric example to xgboost docs

### PR DESCRIPTION
**Pull request by djsauble**
Fixes: https://github.com/iterative/dvclive/issues/721

The XGBoost docs have a dangling `param` variable that isn't defined anywhere. Since this is where you define your metric, and this is an important part of experimentation, I added an example of how to use it to set a metric via the `eval_metric` param.

### CODE REVIEW

**Comment by dberenbaum** [comment]
Thanks @djsauble! As long as we are making changes, can we fix it up a bit more?

For example, can it follow a workflow that looks like the xgboost get started [docs](https://xgboost.readthedocs.io/en/stable/get_started.html#python)? It seems like that's what they see as their primary interface, and subjectively it looks easier to understand to me.

It would also be nice to flesh out the top example even more to show where other variables like `dtrain` and `dval` come from.

**Comment by djsauble** [comment]
> For example, can it follow a workflow that looks like the xgboost get started [docs](https://xgboost.readthedocs.io/en/stable/get_started.html#python)? It seems like that's what they see as their primary interface, and subjectively it looks easier to understand to me.

@dberenbaum Just to clarify, you're saying we should use something like this:

```
model = xgb.XGBClassifier()
model.fit(X_train, y_train)
```

…instead of…

```
xgb.train()
```

?

> It would also be nice to flesh out the top example even more to show where other variables like dtrain and dval come from.

This is the working XGBoost code I ended up with after going through our existing docs + the hint about metrics from @mattseddon:

```
from dvclive import Live
from dvclive.xgb import DVCLiveCallback
from xgboost import XGBClassifier
import xgboost as xgb
from sklearn import datasets
from sklearn.model_selection import train_test_split

iris = datasets.load_iris()

X_train, X_test, y_train, y_test = train_test_split(iris.data, iris.target, train_size=0.8)

dtrain = xgb.DMatrix(X_train, label=y_train)
dtest = xgb.DMatrix(X_test, label=y_test)

with Live("results") as live:
    xgb.train(
        {
            "eval_metric": "rmsle"
        },
        dtrain,
        100,
        early_stopping_rounds=5,
        callbacks=[DVCLiveCallback("eval_data")],
        evals=[(dtest, "eval_data")]
    )

    live.log_metric("summary_metric", 1.0, plot=False)
```

How much of this do you want to see in our docs? I don't really think we should go all the way back to a specific dataset. The focus here should be on the call to `train()` and the associated base parameters.

File: content/docs/dvclive/ml-frameworks/xgboost.md [hunk 1]
```diff
@@ -14,9 +14,16 @@ from dvclive.xgb import DVCLiveCallback
...
+  "eval_metric": "rmse"
+}
+
 xgboost.train(
-    param, dtrain, num_round=5, evals=[(dval, "eval_data")]
+    params,
+    dtrain,
+    num_round=5,
     callbacks=[DVCLiveCallback("eval_data")],
+    evals=[(dval, "eval_data")],
```
Comment (dberenbaum) [inline content/docs/dvclive/ml-frameworks/xgboost.md:26]: Why do we use the name `eval_data` on both of these lines? Since I don't think they represent the same thing, I find it a bit confusing. You didn't introduce this, but I wonder if we should just drop `eval_data` from the `DVCLiveCallback`?

Also, any reason you moved `evals` below `callbacks`?

**Comment by dberenbaum** [comment]
Fair points, @djsauble. I don't want to block the improvements you made, so I just left one minor comment to clarify and otherwise it looks good, thanks!

**Comment by djsauble** [comment]
@dberenbaum I actually agree with you that using `.fit()` makes sense. It's more familiar to me anyway. 🙂

I'll fix this up a bit more and ping you again for a quick review.

**Comment by dberenbaum** [comment]
> @dberenbaum I actually agree with you that using `.fit()` makes sense. It's more familiar to me anyway. 🙂

If we only show that `.fit()` sklearn-like interface, do you think it leaves the user confused about how to do it with `xgb.train()`?

**Comment by djsauble** [comment]
Let's choose one or the other. The sklearn interface is probably more familiar to people, but I don't feel strongly about this. It looks like both interfaces support callbacks, which is the only hard requirement for DVCLive.

Which interface do you want to go with?

**Comment by dberenbaum** [comment]
Sounds good. Let's go with the sklearn interface then.

File: content/docs/dvclive/ml-frameworks/xgboost.md (hunk 1 above)
Comment (djsauble) [reply content/docs/dvclive/ml-frameworks/xgboost.md]: Dropped "eval_data" from DVCLiveCallback() and made `callbacks` to be the last parameter.

**Comment by djsauble** [comment]
@dberenbaum Switched to the sklearn interface and tested all code snippets locally. Think this is good now.

**Review by dberenbaum — APPROVED** [review]
